File: app/src/main/java/th/co/thekhaeng/materialstyletemplate/adapter/item/TopicItem.java
package th.co.thekhaeng.materialstyletemplate.adapter.item;

import android.support.annotation.ColorInt;

import th.co.thekhaeng.materialstyletemplate.adapter.base.BaseItem;

import static th.co.thekhaeng.materialstyletemplate.adapter.ItemCreator.TYPE_TOPIC;

/**
 * Created by thekhaeng on 4/7/2017 AD.
 */

public class TopicItem extends BaseItem{
    private String topic;
    private int color;

    public TopicItem( String topic, @ColorInt int color ){
        super( TYPE_TOPIC );
        this.topic = topic;
        this.color = color;
    }

    public String getTopic(){
        return topic;
    }

    public int getColor(){
        return color;
    }
}

File: app/src/main/kotlin/com/thekhaeng/materialstyletemplate/adapter/item/TopicItem.kt
package com.thekhaeng.materialstyletemplate.adapter.item

import androidx.annotation.ColorInt
import com.thekhaeng.materialstyletemplate.adapter.ItemConstant.TYPE_TOPIC
import com.thekhaeng.materialstyletemplate.adapter.base.BaseItem

/**
 * Created by thekhaeng on 4/7/2017 AD.
 */

class TopicItem(val topic: String,
                @ColorInt val color: Int) : BaseItem(TYPE_TOPIC)
